File: daily_menu/src/test/java/hu/soft4d/yokudlela3/integration/test/v1/cucumber/CucumberTest.java
package hu.soft4d.yokudlela3.integration.test.v1.cucumber;

import hu.soft4d.yokudlela3.integration.BaseIntegrationTest;
import hu.soft4d.yokudlela3.integration.IntegrationTestConstant;

import org.eclipse.microprofile.config.ConfigProvider;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

//@ConfigurationParameter(key = PLUGIN_PUBLISH_QUIET_PROPERTY_NAME, value = "true")
//@IncludeEngines("cucumber")
public class CucumberTest extends BaseIntegrationTest {
    // IMPORTANT NOTE: class name must be start or end "Test" pre or postfix (in case of using those annotations above)

    public static final String ROOT_URI = "root-uri";
    public static final String KEYCLOAK_URI = "keycloak-uri";
    public static final String KEYCLOAK_CLIENT = "keycloak-client";
    public static final String KEYCLOAK_CLIENT_VALUE = "account";
    public static final String TEST_FILES_PATH = "test-files-path";
    public static final String TEST_FILES_PATH_VALUE = "src/test/resources/test_objects/";


    @BeforeAll
    public static void setSystemProperties() {
        String port =
                ConfigProvider.getConfig().getValue("quarkus.http.test-port", String.class);
        System.setProperty(ROOT_URI, IntegrationTestConstant.HOST + ":" + port);
        String authServerUrl =
                ConfigProvider.getConfig().getValue(
                        IntegrationTestConstant.QUARKUS_OIDC_AUTH_SERVER_URL ,
                        String.class)
                         + "/protocol/openid-connect/token";
        System.setProperty(KEYCLOAK_URI, authServerUrl);
        System.setProperty(KEYCLOAK_CLIENT, KEYCLOAK_CLIENT_VALUE);
        System.setProperty(TEST_FILES_PATH, TEST_FILES_PATH_VALUE);
    }

    @Test
    public void run() {
        io.cucumber.core.cli.Main.run(new String[]{"--glue", "hu.soft4d.integration", "classpath:features"});
    }

}

File: kitchen/src/test/kotlin/com/yokudlela/kitchen/cucumber/CucumberTest.kt
package com.yokudlela.kitchen.cucumber

import com.yokudlela.kitchen.Constants.Companion.HTTP_LOCALHOST
import com.yokudlela.kitchen.Constants.Companion.KEYCLOAK_AUTH_SERVER_URL
import com.yokudlela.kitchen.Constants.Companion.KEYCLOAK_CLIENT
import com.yokudlela.kitchen.Constants.Companion.KEYCLOAK_CLIENT_VALUE
import com.yokudlela.kitchen.Constants.Companion.KEYCLOAK_TOKEN_URL
import com.yokudlela.kitchen.Constants.Companion.KEYCLOAK_URI
import com.yokudlela.kitchen.Constants.Companion.ROOT_URI
import com.yokudlela.kitchen.Constants.Companion.TEST_FILES_PATH
import com.yokudlela.kitchen.Constants.Companion.TEST_FILES_PATH_VALUE
import io.cucumber.core.options.Constants.GLUE_PROPERTY_NAME
import io.cucumber.core.options.Constants.PLUGIN_PUBLISH_QUIET_PROPERTY_NAME
import io.cucumber.java.Before
import org.junit.platform.suite.api.ConfigurationParameter
import org.junit.platform.suite.api.IncludeEngines
import org.junit.platform.suite.api.SelectClasspathResource
import org.junit.platform.suite.api.Suite
import org.springframework.boot.test.web.server.LocalServerPort

@Suite
@SelectClasspathResource("features")
@ConfigurationParameter(
    key = GLUE_PROPERTY_NAME,
    value = "classpath:com.yokudlela.integration, classpath:com.yokudlela.kitchen"
)
@ConfigurationParameter(key = PLUGIN_PUBLISH_QUIET_PROPERTY_NAME, value = "true")
@IncludeEngines("cucumber")
class CucumberTest {

    @LocalServerPort
    private var port: Int = 0

    // ! must use cucumber annotation
    @Before
    fun setSystemProperties() {
        System.setProperty(ROOT_URI, "$HTTP_LOCALHOST:$port")
        System.setProperty(KEYCLOAK_URI, System.getProperty(KEYCLOAK_AUTH_SERVER_URL) + KEYCLOAK_TOKEN_URL)
        System.setProperty(KEYCLOAK_CLIENT, KEYCLOAK_CLIENT_VALUE)
        System.setProperty(TEST_FILES_PATH, TEST_FILES_PATH_VALUE)
    }
}